includes/helper/class-wp-job-manager-helper-rest-api.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * File containing the class WP_Job_Manager_Helper_REST_API.
+ *
+ * @package wp-job-manager
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WP_Job_Manager_Helper_REST_API
+ */
+class WP_Job_Manager_Helper_REST_API {
+
+	/**
+	 * The nonce helper to validate the request.
+	 *
+	 * @var WP_Job_Manager_Helper_Nonce
+	 */
+	private WP_Job_Manager_Helper_Nonce $nonce;
+
+	/**
+	 * The namespace.
+	 *
+	 * @var string
+	 */
+	private const NAMESPACE = 'wpjm-internal/v1';
+
+	/**
+	 * Rest base for the current object.
+	 *
+	 * @var string
+	 */
+	private const REST_BASE = '/licensing';
+
+	/**
+	 * Construct the REST API class.
+	 *
+	 * @param WP_Job_Manager_Helper_Nonce $nonce
+	 */
+	public function __construct( WP_Job_Manager_Helper_Nonce $nonce ) {
+		$this->nonce = $nonce;
+	}
+
+	/**
+	 * Initialize the hooks for the REST API.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		add_action( 'rest_api_init', [ $this, 'register_rest_routes' ] );
+	}
+
+	/**
+	 * Register the REST routes related to licensing.
+	 *
+	 * @return void
+	 */
+	public function register_rest_routes() {
+		register_rest_route(
+			self::NAMESPACE,
+			self::REST_BASE . '/receive-wpcom-license-key',
```

Comment:
We could simply name this endpoint `license` and the PUT/POST method will imply that the license is created.